src/vivarium_gates_nutrition_optimization/components/mortality.py

```diff
@@ -0,0 +1,120 @@
+from typing import Union
+
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.lookup import LookupTable
+from vivarium.framework.population import PopulationView, SimulantData
+from vivarium.framework.randomness import RandomnessStream
+from vivarium.framework.values import Pipeline
+
+from vivarium_gates_nutrition_optimization.constants import data_keys
+
+
+class MaternalMortality:
+    def __init__(self):
+        self._randomness_stream_name = "mortality_handler"
+
+        self.cause_of_death_column_name = "cause_of_death"
+        self.years_of_life_lost_column_name = "years_of_life_lost"
+
+        self.mortality_probability_pipeline_name = "mortality_probability"
+
+    @property
+    def name(self) -> str:
+        return "maternal_mortality"
+
+    def __repr__(self) -> str:
+        return f"MaternalMortality()"
+
+        # noinspection PyAttributeOutsideInit
+
+    def setup(self, builder: Builder) -> None:
+        self.random = self._get_randomness_stream(builder)
+        self.clock = builder.time.clock()
+
+        self.mortality_probability = self._get_mortality_probability(builder)
+
+        self.life_expectancy = self._get_life_expectancy(builder)
+
+        self.population_view = self._get_population_view(builder)
+
+        self._register_simulant_initializer(builder)
+        self._register_on_timestep_listener(builder)
+
+    def _get_randomness_stream(self, builder: Builder) -> RandomnessStream:
+        return builder.randomness.get_stream(self._randomness_stream_name)
+
+        # noinspection PyMethodMayBeStatic
+
+    def _get_life_expectancy(self, builder: Builder) -> Union[LookupTable, Pipeline]:
+        life_expectancy_data = builder.data.load(
+            "population.theoretical_minimum_risk_life_expectancy"
+        )
+        return builder.lookup.build_table(life_expectancy_data, parameter_columns=["age"])
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        return builder.population.get_view(
+            [
+                self.cause_of_death_column_name,
+                self.years_of_life_lost_column_name,
+                "alive",
+                "exit_time",
+                "age",
+                "sex",
+                "pregnancy",
+            ]
+        )
+
+    def _get_mortality_probability_source(self, builder: Builder):
```

Comment:
Nit: you don't use this function anywhere and don't persist the lookup table, so you might as well just build it inside `_get_mortality_probability`.